### ISSUE TITLE
`--php-version` option doesn't warn about invalid syntax

### ISSUE THREAD

**Issue by danielbeardsley** [vimeo/psalm#3326]
I expected psalm to warn me about trying to use a syntax of php that's not supported by my current version. A bunch of new php-7.4 features aren't warned about when selecting php versions that don't support them:

```php
<?
class TypedProp {
   public int $apple = 0;
}

$arrow = fn(int $y): int => $y+1;

$c ??= 120;
```
Running psalm on that:
```
▶  psalm --php-version=5.6 ./7.4.php
Scanning files...
Analyzing files...

░

------------------------------
No errors found!
------------------------------

Checks took 0.19 seconds and used 34.401MB of memory
Psalm was unable to infer types in the codebase
```

**Comment by danielbeardsley** [comment]
> can you reproduce the issue on https://psalm.dev ?

I tried, but there is no way to change the `--php-version` option there.

**Comment by muglug** [comment]
I could gate certain things, but ultimately Psalm isn't going to be a good way to prevent this – recent versions of PHP Parser allow things (like commas in argument lists) that will break in 5.6 in a manner largely undetectable by Psalm.

Instead you should run `php -l` on your files.
